libs/data-mapper/src/lib/core/state/DataMapSlice.ts

```diff
@@ -619,6 +619,53 @@ export const deleteConnectionFromConnections = (connections: ConnectionDictionar
   );
 };
 
+export const deleteParentRepeatingConnections = (connections: ConnectionDictionary, inputKey: string /* contains prefix */) => {
+  const parentId = getParentId(inputKey);
+  if (parentId.endsWith('source-')) {
```

Comment:
I wanna say there's a const for 'source-' somewhere in either ReactFlow Utils or Constants